server/main.go

```diff
@@ -19,11 +19,52 @@ var dagList = make(map[string]*dag.Dag)
 
 func main() {
 	http.HandleFunc("/dag/add", addDagHandler)
+	http.HandleFunc("/dag/schedule", scheduleDagHandler)
 	fmt.Println("Server listening on :8080")
 	http.ListenAndServe(":8080", nil)
 }
 
-// Handler function for the "/api/hello" endpoint
+func scheduleDagHandler(w http.ResponseWriter, r *http.Request) {
+	clientId := r.Header.Get("Client-Id")
+	if clientId == "" {
+		http.Error(w, "Client-Id header is missing", http.StatusBadRequest)
+		return
+	}
+
+	dagName := r.Header.Get("dag-name")
+	if dagName == "" {
+		http.Error(w, "dag-name header is missing", http.StatusBadRequest)
+		return
+	}
```

Comment:
1. Query param instead of header param
2. Before printing Error, let's log them too. log.Printf("...")